python/tank/platform/framework.py

```diff
@@ -25,12 +26,52 @@
 from ..util import log_user_activity_metric
 
 # global variable that holds a stack of references to
-# a current bundle object - this variable is populated
+# a current bundle object, per thread - this variable is populated
 # whenever the bundle.import_module method is executed
 # and is a way for import_framework() to be able to resolve
 # the current bundle even when being recursively called 
-# inside an import_module call
-CURRENT_BUNDLE_DOING_IMPORT = []
+# inside an import_module call.
+__threadlocal_storage = threading.local()
+
+
+def _get_thread_import_stack():
```

Comment:
I think these new methods should be in a separate file. Maybe makes sense to wrap up in a class too instead of that  `__threadlocal_storage` member?